core/src/main/groovy/noe/tomcat/configure/ConnectorConfiguratorTomcat.groovy

```diff
@@ -124,29 +124,33 @@ class ConnectorConfiguratorTomcat {
   }
 
   private void updateExistingConnector(Node connector, Node newConnector) {
-    // update attributes
+    // update connector attributes
     newConnector.attributes().each { attribute ->
       connector.@"${attribute.key}" = attribute.value
     }
 
-    // update nodes
     newConnector.each { Node newSubelement ->
-      String UpgradeProtocol = ConnectorUpgradeProtocolTomcat.ELEMENT_NAME
-
-      if (newSubelement.name() == UpgradeProtocol) {
-        if (connector.find { it.name() == UpgradeProtocol } == null) {
-          // create new element
-          connector.appendNode(newSubelement, newSubelement.attributes(), newSubelement.value())
-        } else {
-          // upgrade existing element
-          connector.findAll { it.name() == UpgradeProtocol }.each { upgradeProtocol ->
-            newSubelement.attributes() { attribute ->
-              upgradeProtocol.@"${attribute.key}" = attribute.value
-            }
-          }
-        }
+      updateInnerElements(connector, newSubelement)
+    }
+  }
+
+  /**
+   *
+   * @param connector
+   * @param newSubElement
+   *
+   * Search for existing inner element to remove and replace with new one
+   * Note: SSLHostConfig inner elements can exist more than one
+   */
+  private void updateInnerElements(Node connector, Node newSubElement) {
+    if (connector.find { it.name() == newSubElement.name() } != null) {
+      Node oldElement = (Node) connector.find { it.name() == newSubElement.name()}
+      if (oldElement.name() != "SSLHostConfig") {
+        connector.remove(oldElement)
```

Comment:
Remove all subelements but `SSLHostConfig` ? So if somebody has some custom configuration (reproducer provided by customer) with upgradeProdocal set already and we want to add some configuration wrt SSLHostConfig, will the upgradeProtocal configuration disappear automagically? Or am I missing something?

Reply:
As the new configuration will have SSLHostConfig element, the UpgradeProtocol will stand its ground.The new SSLHostConfig element will be added along with the old. The removing is not related with SSLHostConfig element because someone might want to have multiple SSLHostConfig elements inside a connector.I have seen configurations in which it's used like that. Am i right?